src/allmydata/storage/http_server.py

```diff
@@ -527,27 +544,47 @@ def read_data(offset: int, length: int) -> bytes:
             # https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3861
             raise _HTTPError(http.NOT_ACCEPTABLE)
 
-    def _read_encoded(self, request, schema: Schema) -> Any:
+    def _read_encoded(
+        self, request, schema: Schema, max_size: int = 1024 * 1024
```

Comment:
This is some scope creep, but maybe the other call sites could pick their own max size now and `_read_encoded` can skip providing a default value.  For example, the `allocate_buckets` message should probably never be more than 1kB (limited number of shares, and you'd need an absolutely absurdly huge file for the encoded size to get even close to 1kB).  whereas `advise_corrupt_share_immutable` accepts some semi-arbitrary text so being a bit more generous and accepting something like 64kB makes sense.

Again, this is scope creep, feel free not to bother right now.

Reply:
I'll file new ticket.